# Confidence Interval

### Why Confidence Intervals?

Confidence intervals are a calculated range or boundary around a parameter or a statistic that is supported mathematically with a certain level of confidence.  For example, in the lecture, we estimated, with 95% confidence, that the population proportion of parents with a toddler that use a car seat for all travel with their toddler was somewhere between 82.2% and 87.7%.

This is *__different__* than having a 95% probability that the true population proportion is within our confidence interval.

Essentially, if we were to repeat this process, 95% of our calculated confidence intervals would contain the true proportion.

### How are Confidence Intervals Calculated?

Our equation for calculating confidence intervals is as follows:

$$Best\ Estimate \pm Margin\ of\ Error$$

Where the *Best Estimate* is the **observed population proportion or mean** and the *Margin of Error* is the **t-multiplier**.

The t-multiplier is calculated based on the degrees of freedom and desired confidence level.  For samples with more than 30 observations and a confidence level of 95%, the t-multiplier is 1.96

The equation to create a 95% confidence interval can also be shown as:

$$Population\ Proportion\ or\ Mean\ \pm (t\ multiplier *\ Standard\ Error)$$

Lastly, the Standard Error is calculated differenly for population proportion and mean:

$$Standard\ Error \ for\ Population\ Proportion = \sqrt{\frac{Population\ Proportion * (1 - Population\ Proportion)}{Number\ Of\ Observations}}$$

$$Standard\ Error \ for\ Mean = \frac{Standard\ Deviation}{\sqrt{Number\ Of\ Observations}}$$

In [7]:
n_case = 659
positive_case = 540

p_hat = 540/659

# 95 percent
confi_interval = p_hat + 1.96*((p_hat*(1-p_hat))/n_case)**.5

In [8]:
p_hat - 1.96*((p_hat*(1-p_hat))/n_case)**.5

0.7900537499137914

In [10]:
p_hat + 1.96*((p_hat*(1-p_hat))/n_case)**.5

0.8487929875672404

Different z Multiplier

- 90% = 1.645
- 95% = 1.96
- 98% = 2.326
- 99% = 2.576


# Conservativs confedence interval

In [1]:
n_case = 659
positive_case = 540

p_hat = 540/659

In [9]:
p_hat + (1/(n_case)**0.5)

0.8583778380970969

In [7]:
p_hat - (1/(n_case)**0.5)

0.7804688993839349

# Difference in proportion Confidence Interval

Best Estimate $\pm$ Margin of Error

$\hat{P}_1$ - $\hat{P}_2$ $\pm$ Margin of Error<br>
$\hat{P}_1$ - $\hat{P}_2$ $\pm$ "a few" * se($\hat{P}_1$ - $\hat{P}_1$) <br>
$\hat{P}_1$ - $\hat{P}_2$ $\pm$ 1.96 * 
 $\sqrt{\frac{ \hat{P}_1(1- \hat{P}_1)}{n_1} + \frac{ \hat{P}_2(1- \hat{P}_2)}{n_2}}$

What is the difference in population of parents reporting that their childer age 6-18 have some swiming lessons between white children and black children?<br>

- Difference in population proportions($\hat{P}_1$ - $\hat{P}_2$)

Survey report

- out of 247 parents of black children  91 says their childnre were taking swiming lessons age 6-18. 
- out of 988 white children  543 says their childnre were taking swiming lessons age 6-18.




In [23]:
n1 = 988
n2 = 247
p_hat1 = 543/988  # parents of white
p_hat2 = 91/247   # parents of black

p_hat1 - p_hat2

0.18117408906882598

In [29]:
DIff_prop = 0.18 + 1.96*((p_hat1*(1-p_hat1)/(n1)) +
                         (p_hat2*(1-p_hat2)/(n2)))**.5

DIff_prop

0.2476867047470739

In [30]:
Diff_prop = 0.18 - 1.96*((p_hat1*(1-p_hat1)/(n1)) +
                         (p_hat2*(1-p_hat2)/(n2)))**.5

Diff_prop

0.11231329525292608

with 95% confidence, the population proportion of parents with white children who have taken swimming lessons is 11.23 to 24.77% higher than the population proportion of parents with black children who have taken swimming lessons

### interval for difference
- is there a difference between two parameters ?
- if paramters are equal --> difference is 0
- if parameters are unequal --> difference is not 0<br>

Look for 0 in range of resonable values, if zero is not contqained in that confidence interval then our parameters are unequal

### Asumptions
- we need to assume that we have two independent random samples.

we also need large enough sample sizes to assume that the distribution of out estimate is normal. that is we need $n_1\hat{P}_1, n_1(1-\hat{P}_2), n_2\hat{P}_2$ and $n_2(1-\hat{P}_2)$ to all be at least 10

- which means we need atleast 10 Yes and 10 No for each sample


## Closer Look at estimated SE
Estimated Standard Error = $\sqrt{\frac{\hat p(1-\hat p)}n}$

which is maximized when **${\hat p}$ = 0.5**

conservative standard error = $\frac{1}{2*\sqrt{n}}$

Example<br>
$\hat p$ $\pm$ 2 * $\frac{1}{2\sqrt{n}}$ <br>
=> $\hat p$ $\pm$  $\frac{1}{\sqrt{n}}$ 

$\hat p$ = 0.85<br>
n = 659

- Conservative 95% Confidence Interval, 4% Margin of Error

In [35]:
print(round(0.85-(1/(659)**0.5),2),round(0.85+(1/(659)**0.5),2))

0.81 0.89


### MOE

MOE = $\frac{1}{\sqrt n}$

MOE depends on confidence level and sample size

**What sample size would we need** to have `95 percent (conservative) confidene interval` with a Margin of `Error` of only `3%` (0.03)?

In [25]:
n = 1/(0.03)**2
n

1111.111111111111

- sample size is inversely proportional to margin error
- sample size should be greater then 1112 to have hmargin error of 3 %.
- n>= 1,112

### change the confidence inteval to 99 %

- $\hat P$ $\pm$ Z* $\frac{1}{(2 \sqrt n}$)

- **MoE** = Z * $\frac{1}{2 \sqrt n}$

- n = ($\frac {Z^* }{ 2*MoE)}$)$^2$

In [12]:
# at 99 per condidence z = 2.576

n = (2.576/(2*0.03))**2
n

1843.2711111111114

In [27]:
# at 94 per condidence z = 1.96 , moe = 4% 
n = (1.96/(2*0.04))**2
n

600.25

In [28]:
n = (2.326/(2*0.03))**2
n

1502.8544444444444

In [29]:
1.96 *(1/(2*(np.sqrt(232))))

0.06434015210126405

                   Summary 
- Estimated standard error may be too small, or inaccurate based off sample so cam employ conservative approch.

- Conservative approch --> determine sample size needed based on confidence level and desired margin error

n>=1844

In [14]:
(1.96/(2*0.03))**2

1067.1111111111109

# One Mean

### Sample Distribuation of sample Mean

If model for popoulation of response is approximately normal (or sample size is 'large' engough), distribution of sample mean is (approx.) normal. 

Standard deviation of sampling distribution of statistics is called s standard error of that statistc.
- Standard error of the sample mean = $\frac {\sigma}{\sqrt n}$

- Estimated standard error of sample mean = $\frac {S}{\sqrt n}$

### Confidence Interval Basics

**Best Estiamte** $\pm$ Margin of Error<br>
**Best Estimate** = Unbiased Point Estimate<br>
**Margin OF Error** = "a few" Estimate Standard Errors<br>
**"a few"** = multiplier from appropriate distribution based on desired confidence level and sample design<br>

**95% Confidence Level = 0.05 Significance**<br>

sample mean $\pm$ "a few" * estimate standard error of sample mean<br>
### $\bar x$ $\pm$ $t^*$ $\frac{s}{\sqrt n}$

$t^*$ multiplier comes from a t-distribution with n-1 degree of freedom.<br>
95% confidence<br>
n = 25 --> $t^*$ = 2.064<br>
n = 1000 --> $t^*$ = 1.962<br>

if we do have 95 % confidence and we have sample size of 25 as we do here then multiplier have to be litter bit more than 2.064 in order to have that 95% confidence level attained.<br>

With a larger sample size, our sample standard deviation is going to be better estimate of that true standrd deviation and our multiplier will start to reflect that.

In [43]:
Mean = 82.48
Sd = 15.06
n = 25
t = 2.064

print(round(82.48 + 2.064 *(15.06/(25)**.5),2),
      round(82.48 - 2.064 *(15.06/(25)**.5)),2)

88.7 76 2


With 95% confidence, the population mean cartwheel distance for all adults is estimated to be between 76.26 inches and 88.80 inches.<br>

#### What does "with 95% confidence" mean?<br>

if this procedure were reapeted over and over each time producing a 95% confidence interval estimate,<br>

we would **expect 95% of those resulting intervals to contain the popuplation mean cartwheel distance.**<br>

                    Summary
- Confidence Interval are used to give an interval estimate for our parameter of interest ~ **a population mean.**

- Center of Confidence Interval is our best estiamte ~ **the sample mean.**

- Margin of Error is "few" (estimate) standard errors **~ for means we use $t^*$ multipliers**

## Difference in means for paired Data

- When we have paired data, we want to treat the two sets of values simultaneously. <br>
- We can't look at them separately because they're matched in some way.
  - Measurement colleted on the same individual.
  - Measurement colleted on the matched individual.
    - age, gender, weight.
- variable: Difference of measurement within pairs.

              Reaserch Question
              
what is the average difference between the older twin's and younger twin's self-reported education?

- Population - All identical twins.
- Parameter of Interest- Population mean difference of self report education level $\mu_d$.
              diffence = older - younger
              
Construct 95% confidence Interval for mean difference of self-reported education for a set of identical twins.<br>

n = 340 observations<br>
Minimum = -3.5 years<br>
maximum = 4 years<br>
72.1% had a difference of 0 years<br>

Mean = 0.0838 years<br>
Standard Deviation = 0.7627 years.


**Best Estiamte** $\pm$ Margin of Error<br>

sample mean difference $\pm$ "a few" * estimate standard error<br>

### $\bar x$ $\pm$ $t^*$ $\frac{S_d}{\sqrt n}$

$t^*$ multiplier comes from a t-distribution with n-1 degree of freedom.<br>
95% confidence<br>
n = 25 --> $t^*$ = 2.064<br>
n = 1000 --> $t^*$ = 1.962<br>

In [64]:
n = 340 
t = 1.967
x = 0.084
t = 1.967
sd= 0.76

print(round(x - t*(sd/(n)**.5),4),
      round(x + t*(sd/(n)**.5),4))

0.0029 0.1651


In [ ]:
n = 500 
t = 1.967
x = 0.084
t = 1.967
sd= 0.76

print(round(x - t*(sd/(n)**.5),4),
      round(x + t*(sd/(n)**.5),4))

            Interpretation the confidence Interval

**"range of resonable values for our parameter"**

With 95% confidence, the population mean difference of the older tiwen's less the Young twin's self-reported education is estimated to be between 0.0025 years and 0.1652

#### How to interpret that confidence interval
Confidence interval are range of resonable values for our parameter. So keeping in mind that we go from 0.0025 years to 0.1652 years, we seee that both ends of our confidence interval are on the positive side. That mean all the values within the confidence interval are on the positive side. That indicate that with 95 percent confidence we actually expect older twin to ahve had more education since a positive value incidaces that the older twin does have more education. However these values are close to zero so we can say the education level jump for the twin migh not be a huge increase.

### Interval for Difference

- is there a mean differene between the eduation level of twins?
-  If education are general equal --> mean difference is 0
- If education levels are unequal --> mean difference is not 0
- Look for 0 in the range of resonable values


### Asumptions

We need to asume the we have **random sample of identical twins sets.**

**Population of differences is normal** (or a large enough sample size can help to bypass the assumption)

### Summary
 - Extension of the mean confidence interval<br>
   ~ use difference variable now.
 - Data need to be paired to calculate a difference variable<br>
   ~ two measurements on same individual<br>
   ~ two measurements on similar, matched individuals.

# Estimating a Difference in Population Means with Confidence

                Research Question
                
Consider Mexican - American adults(ages 18 - 29) living in United States, do males and females differ significantly in mean Body Mas Index (BMI)?

- **Population Mexican** - American (age 18-19) in U.S.
- **Parameter of Interest**($\mu_1$ - $\mu_2$): Body Mass Index or BMI (kg/$m^2$)


### Sampling Distribution of Difference in Two (Independent) Sample Means

- Standard error  =
$\sqrt{\frac{\sigma_1^2}{n_1} + \frac{\sigma_1^2}{n_2}}$

- Estimated standard  = $\sqrt{\frac{S_1^2}{n_1} + \frac{S_1^2}{n_2}}$


             Confidence Interval approches
             
                  Pooled Approach
The variance of two population are assumed to be equal ($\sigma_1^2$ = $\sigma_2^2$)<br>

                  Unpooled Approach
The assumption of equal variance is dropped
 
### Unpooled Confidence Interval calculations
Best Estimate $\pm$ Margin of Error

Difference in sample means $\pm$ "a few" * estimated standard error

($\bar x_1$ - $\bar x_2$) $\pm$ $t^*$ $\sqrt{\frac{S_1^2}{n_1} + \frac{S_1^2}{n_2}}$

### Pooled Confidence Interval calculations
Best Estimate $\pm$ Margin of Error

Difference in sample means $\pm$ "a few" * estimated standard error

($\bar x_1$ - $\bar x_2$) $\pm$  $t^*$ $\sqrt{\frac{(n_1-1)s_1^2 + (n_2-1)s_2^2}{n_1+n_2 -2}}  \sqrt{\frac{1}{n_1} + \frac{1}{n_2}}$
 
$t^*$ multiplier comes from t- distribution with $n_1$ + $n_2$ -  2 degree of freedom. Again, this `approch can be used if we assume the population variaces are equal`.

In [73]:
# Male
mean_m = 23.57
sd_m = 6.24
n_m = 258

#Female
mean_f = 22.83
sd_f = 6.43
n_f = 239

(mean_m - mean_f) - 1.98 *((((258-1)*6.24**2)+((239-1)*6.43**2))/(258 + 239 -2))**.5 * (1/258 +1/239)**.5


-0.38558930273222525

In [74]:
(mean_m - mean_f) + 1.98 * ((((258-1)*6.24**2)+((239-1)*6.43**2))/(258 + 239 -2))**.5* (1/258 +1/239)**.5


1.8655893027322292

With 95% confidence, the difference in mean body mass index between males and femaels for all Mexican - American adults (ages 18-29) in the U.S. estimated to be between -0.38 km/$m^2$ and 1.865 km/$m^2$.

we would expect 95% of those resulting intervals to contain the difference in population mean BMI.

                   Summary
- Confidence Intervals are used to give an interval estimate for our parameter of interest<br>
   ~ difference in population means
- Center of the confidence Interval is our best estimate<br>
   ~ difference in sample means.
- Margin of Error is a "few" (estimated) standard errors <br>
   ~ for two means we use $t^*$ multipliers (pooled vs. unpooled)
- Assumptions for CI's for Difference in Population Means<br>
   ~ data are two simple random samples, independence.<br>
   ~ both population of response are normal( else n large helps)

# Confidence Intervals in Python

### How are Confidence Intervals Calculated?

Our equation for calculating confidence intervals is as follows:

$$Best\ Estimate \pm Margin\ of\ Error$$

Where the *Best Estimate* is the **observed population proportion or mean** and the *Margin of Error* is the **t-multiplier**.

The t-multiplier is calculated based on the degrees of freedom and desired confidence level.  For samples with more than 30 observations and a confidence level of 95%, the t-multiplier is 1.96

The equation to create a 95% confidence interval can also be shown as:

$$Population\ Proportion\ or\ Mean\ \pm (t-multiplier *\ Standard\ Error)$$

The Standard Error(SE) is calculated differenly for population proportion and mean:

$$Standard\ Error \ for\ Population\ Proportion = \sqrt{\frac{Population\ Proportion * (1 - Population\ Proportion)}{Number\ Of\ Observations}}$$


$$Standard\ Error \ for\ Mean = \frac{Standard\ Deviation}{\sqrt{Number\ Of\ Observations}}$$


Lastly, the standard error for difference of population proportions and means is:

$$Standard\ Error\ for\ Difference\ of\ Two\ Population\ Proportions\ Or\ Means = \sqrt{(SE_{\ 1})^2 + (SE_{\ 2})^2}$$

In [3]:
import numpy as np

In [4]:
tstar = 1.96
p = .85
n = 659

se = np.sqrt((p * (1 - p))/n)
se

0.01390952774409444

In [5]:
lcb = p - tstar * se
ucb = p + tstar * se
(lcb, ucb)

(0.8227373256215749, 0.8772626743784251)

In [7]:
import statsmodels.api as sm

sm.stats.proportion_confint(n * p, n)

(0.8227378265796143, 0.8772621734203857)

In [10]:
import pandas as pd
df = pd.read_csv('/Users/knight/Desktop/language/python/practice/Dataset/auto-mpg.csv')
df.head(2)

mpg  cylinders  displacement horsepower  weight  acceleration  model year  \
0  18.0          8         307.0        130    3504          12.0          70   
1  15.0          8         350.0        165    3693          11.5          70   

   origin                   car name  
0       1  chevrolet chevelle malibu  
1       1          buick skylark 320

In [13]:
mean = df["mpg"].mean()
sd   = df["mpg"].std()
n    = len(df)
n 

398

**t - multiplier**  is going to be standrd 1.96
- if number of obeservation is < 30 then we have to look at t-table with degree of freedom = (n-1) and confidence level 95 percent.

In [14]:
tstar = 1.96

se = sd/np.sqrt(n)
se

0.3917798927426896

In [15]:
lcb = mean - tstar * se
ucb = mean + tstar * se
(lcb, ucb)

(22.74668427454594, 24.282461454097287)

In [17]:
sm.stats.DescrStatsW(df["mpg"]).zconfint_mean()

(22.74669838467897, 24.282447343964243)

In [23]:
tstar = 1.96
p = .43
n = 232

se = np.sqrt((p * (1 - p))/n)
se

0.03250331548075334

# Creating confidence intervals in python

IMPORTANT NOTE: In this assessment, you will need to calculate a standard deviation. The Numpy and Pandas libraries both have functions/methods to perform this calculation. But they use different values for a parameter called the Delta Degrees of Freedom or DDOF.

The appropriate value of DDOF to use for this assignment is 1.

Numpy

The numpy.std() function uses a default DDOF of 0. So if you use numpy.std() then you need to set the DDOF parameter to 1.

https://numpy.org/doc/stable/reference/generated/numpy.std.html

Pandas

The pandas.dataframe.std() and pandas.dataframe.describe() methods use a default DDOF of 1.

https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.std.html

In [18]:
import pandas as pd, numpy as np

# Imort the data (uncomment the line below and use this if you downloaded the Jupyter notebook)
df = pd.read_csv("https://raw.githubusercontent.com/UMstatspy/UMStatsPy/master/Course_2/nap_no_nap.csv")
df.head(2)

id     sex  age (months)  dlmo time  days napped  napping  \
0   1  female          33.7      19.24            0        0   
1   2  female          31.5      18.27            0        0   

   nap lights outl time  nap sleep onset  nap midsleep  nap sleep offset  ...  \
0                   NaN              NaN           NaN               NaN  ...   
1                   NaN              NaN           NaN               NaN  ...   

   sleep onset latency  night midsleep time  night wake time  \
0                 0.23                 1.92             7.17   
1                 0.25                 1.09             6.69   

   night sleep duration  night time in bed  24 h sleep duration  \
0                 629.4              643.0                629.4   
1                 672.4              700.4                672.4   

   bedtime phase difference  sleep onset phase difference  \
0                     -1.21                         -1.44   
1                     -0.96                         -1.21   

   midsleep phase difference  wake time phase difference  
0                       6.68                       11.93  
1                       6.82                       12.42  

[2 rows x 25 columns]

**Question**: What variable is used in the column 'napping' to indicate a toddler takes a nap?  
**Question**: What is the sample size $n$? What is the sample size for toddlers who nap, $n_1$, and toddlers who don't nap, $n_2$?

#### Average bedtime confidence interval for napping and non napping toddlers
Create two 95% confidence intervals for the average bedtime, one for toddler who nap and one for toddlers who don't.

Before any analysis, we will convert 'night bedtime' into decimalized time.

In [2]:
# Convert 'night bedtime' into decimalized time
df.loc[:,'night bedtime'] = np.floor(df['night bedtime'])*60 + np.round(df['night bedtime']%1,2 )*100


Now, isolate the column 'night bedtime' for those who nap into a new variable, and those who didn't nap into another new variable.

In [3]:
bedtime_nap = df['night bedtime'][df['napping']==1]

bedtime_no_nap = df['night bedtime'][df['napping']==0]

Now find the sample mean & standard deviation bedtime for nap and no_nap.

In [4]:
nap_mean_bedtime = bedtime_nap.mean()

no_nap_mean_bedtime = bedtime_no_nap.mean()

In [5]:
nap_mean_bedtime - no_nap_mean_bedtime

42.066666666666606

In [16]:
nap_sd_bedtime = bedtime_nap.std()
print('nap_sd_bedtime',nap_sd_bedtime)

no_nap_sd_bedtime = bedtime_no_nap.std()
print('no_nap_sd_bedtime',no_nap_sd_bedtime)

nap_sd_bedtime 34.445540177143954
no_nap_sd_bedtime 34.30014577228499


Now find the standard error for $\bar{X}_{nap}$ and $\bar{X}_{no\ nap}$.

In [116]:
n1 = len(bedtime_nap)
print(n1)
n2 = len(bedtime_no_nap)
print(n2)

15
5


In [117]:
nap_se_bedtime = nap_sd_bedtime/np.sqrt(n1)
nap_se_bedtime

8.893800230479801

In [118]:
no_nap_se_bedtime =  no_nap_sd_bedtime/np.sqrt(n2)
no_nap_se_bedtime

15.339491516996253

We expect the variance in sleep time for toddlers who nap and toddlers who don't nap to be the same. So we use a pooled standard error.

Calculate the pooled standard error of $\bar{X}_{nap} - \bar{X}_{no\ nap}$ using the formula below.

$s.e.(\bar{X}_{nap} - \bar{X}_{no\ nap}) = \sqrt{\frac{(n_1-1)s_1^2 + (n_2-1)s_2^2}{n_1+n_2-2}(\frac{1}{n_1}+\frac{1}{n_2})}$

In [17]:
np.sqrt(((14*8.9**2)+(4*15.33**2))/(1/15+1/5))

87.65648007991194

In [19]:
87.65.cdf(bedtime_nap, df)

AttributeError: 'float' object has no attribute 'cdf'

To build a 95% confidence interval, what is the value of t\*?  You can find this value using the percent point function: 
```
from scipy.stats import t

t.ppf(probabiliy, df)
```
This will return the quantile value such that to the left of this value, the tail probabiliy is equal to the input probabiliy (for the specified degrees of freedom). 

**Question**: What are the 95% confidence intervals, rounded to the nearest ten, for the average bedtime (in decimalized time) for toddlers who nap and for toddlers who don't nap? 

CI = $\bar{X} \pm \ t^* \cdot s.e.(\bar{X})$

Example: to find the $t^*$ for a 90% confidence interval, we want $t^*$ such that 90% of the density of the $t$ distribution lies between $-t^*$ and $t^*$.

Or in other words if $X \sim t(df)$:

P($-t^*$ < X < $t^*$) = .90

Which, because the $t$ distribution is symmetric, is equivalent to finding $t^*$ such that:  

P(X < $t^*$) = .95

So the $t^*$ for a 90% confidence interval, and lets say df=10, will be:

t_star = t.ppf(.95, df=10)


In [6]:
from scipy.stats import t

# Find the t_stars for the 95% confidence intervals
nap_t_star = t.ppf(.95, df=14)
print(nap_t_star)

1.7613101357748562


In [7]:
no_nap_t_star = t.ppf(.95, df=4)
print(no_nap_t_star)

2.13184678133629


In [15]:
import scipy.stats.distributions as dist

pvalue = 2*dist.norm.cdf(-np.abs(nap_t_star))
round(pvalue,4)
pvalue

0.07818592546022286

**Quesion**: What is $t^*$ for nap and no nap?

Now to create our confidence intervals. For the average bedtime for nap and no nap, find the upper and lower bounds for the respective confidence intervals.

In [121]:
lcb = nap_mean_bedtime - nap_t_star * nap_se_bedtime
ucb = nap_mean_bedtime + nap_t_star * nap_se_bedtime
(lcb, ucb)

(1217.4019261751657, 1248.7314071581675)

In [122]:
lcb = no_nap_mean_bedtime - no_nap_t_star * nap_se_bedtime
ucb = no_nap_mean_bedtime + no_nap_t_star * nap_se_bedtime
(lcb, ucb)

(1172.0397806048036, 1209.9602193951964)

In [140]:
import statsmodels.api as sm

sm.stats.DescrStatsW(bedtime_nap).zconfint_mean()


(1215.635138529232, 1250.498194804101)

In [141]:
sm.stats.DescrStatsW(bedtime_no_nap).zconfint_mean()


(1160.9351490855297, 1221.0648509144703)